src/osp/scientific/kepler.h

```diff
@@ -0,0 +1,161 @@
+/**
+ * Open Space Program
+ * Copyright © 2019-2022 Open Space Program Project
+ *
+ * MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+#pragma once
+
+#include "osp/core/math_types.h"
+
+#include <optional>
+
+namespace osp
+{
+
+/**
+ * @brief Holds parameters to fully define a Keplerian (ideal 2-body) orbit
+ */
+struct KeplerOrbit
+{
+private:
+    static constexpr double KINDA_SMALL_NUMBER = 1.0E-8;
+    // These are private as we may want to change the internal representation in the future
+    double m_semiMajorAxis;
+    double m_eccentricity;
+    double m_inclination;
+    double m_argumentOfPeriapsis;
+    double m_longitudeOfAscendingNode;
+    double m_meanAnomalyAtEpoch;
+    double m_epoch;
+    double m_r0;
+    double m_v0;
+    double m_h;
+
+    union
+    {
+        double m_F0;
+        double m_E0;
```

Comment:
can you elaborate on why you have a union of two variables of the same type? 

doesn't seem like this would do any good.

Reply:
It was due to a notational difference between hyperbolic and elliptic orbits, but the variable name would be expressed better as m_eccentricAnomalyAtEpoch regardless.